integration/library_test.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/aquasecurity/fanal/types"
+)
+
+func TestFanal_Library_DockerMode(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		imageName            string
+		imageFile            string
+		expectedFiles        []string
+		expectedOS           analyzer.OS
+		expectedPkgsFromCmds []analyzer.Package
+		expectedLibraries    map[analyzer.FilePath][]godeptypes.Library
+	}{
+		{
+			name:              "happy path, alpine:3.10",
+			imageName:         "alpine:3.10",
+			imageFile:         "testdata/fixtures/alpine-310.tar.gz",
+			expectedOS:        analyzer.OS{Name: "3.10.2", Family: "alpine"},
+			expectedFiles:     []string{"etc/alpine-release", "etc/os-release", "lib/apk/db/installed", "/config"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
+		{
+			name:              "happy path, amazonlinux:2",
+			imageName:         "amazonlinux:2",
+			imageFile:         "testdata/fixtures/amazon-2.tar.gz",
+			expectedFiles:     []string{"etc/system-release", "var/lib/rpm/Packages", "etc/os-release", "/config"},
+			expectedOS:        analyzer.OS{Name: "2 (Karoo)", Family: "amazon"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
+		{
+			name:              "happy path, debian:buster",
+			imageName:         "debian:buster",
+			imageFile:         "testdata/fixtures/debian-buster.tar.gz",
+			expectedFiles:     []string{"var/lib/dpkg/status", "etc/debian_version", "etc/os-release", "usr/lib/os-release", "/config"},
+			expectedOS:        analyzer.OS{Name: "10.1", Family: "debian"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
+		{
+			name:              "happy path, photon:1.0",
+			imageName:         "photon:1.0-20190823",
+			imageFile:         "testdata/fixtures/photon-10.tar.gz",
+			expectedFiles:     []string{"var/lib/rpm/Packages", "etc/lsb-release", "etc/os-release", "/config", "usr/lib/os-release"},
+			expectedOS:        analyzer.OS{Name: "1.0", Family: "photon"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
+		{
+			name:              "happy path, registry.redhat.io/ubi7",
+			imageName:         "registry.redhat.io/ubi7",
+			imageFile:         "testdata/fixtures/ubi-7.tar.gz",
+			expectedFiles:     []string{"etc/redhat-release", "etc/system-release", "/config", "var/lib/rpm/Packages", "etc/os-release"},
+			expectedOS:        analyzer.OS{Name: "7.7", Family: "redhat"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
+		{
+			name:              "happy path, opensuse leap 15.1",
+			imageName:         "opensuse/leap:latest",
+			imageFile:         "testdata/fixtures/opensuse-leap-151.tar.gz",
+			expectedFiles:     []string{"usr/lib/os-release", "usr/lib/sysimage/rpm/Packages", "/config", "etc/os-release"},
+			expectedOS:        analyzer.OS{Name: "15.1", Family: "opensuse.leap"},
+			expectedLibraries: map[analyzer.FilePath][]godeptypes.Library{},
+		},
```

Comment:
Would you add the following image? The image has libraries. We should test a case when `expectedLibraries` is not empty.
https://github.com/aquasecurity/trivy-test-images/blob/master/vulnimage.tar.gz